magicctapipe/scripts/lst1_magic/lst1_magic_stereo_reco.py

```diff
@@ -218,123 +205,113 @@ def stereo_reconstruction(
-                length=u.Quantity(df_tel["length"].iloc[0], u.m),
-                width=u.Quantity(df_tel["width"].iloc[0], u.m),
-                psi=Angle(df_tel["psi"].iloc[0], u.deg),
-                skewness=float(df_tel["skewness"].iloc[0]),
-                kurtosis=float(df_tel["kurtosis"].iloc[0]),
+                intensity=float(df_tel["intensity"]),
+                x=u.Quantity(df_tel["x"], u.m),
+                y=u.Quantity(df_tel["y"], u.m),
+                r=u.Quantity(df_tel["r"], u.m),
+                phi=Angle(df_tel["phi"], u.deg),
+                length=u.Quantity(df_tel["length"], u.m),
+                width=u.Quantity(df_tel["width"], u.m),
+                psi=Angle(df_tel["psi"], u.deg),
+                skewness=float(df_tel["skewness"]),
+                kurtosis=float(df_tel["kurtosis"]),
             )
 
-            image_params = ImageParametersContainer(hillas=hillas_params)
-            event.dl1.tel[tel_id].parameters = image_params
+            event.dl1.tel[tel_id].parameters = ImageParametersContainer(
+                hillas=hillas_params
+            )
 
-        # Reconstruct the stereo parameters:
+        # Reconstruct the stereo parameters
         hillas_reconstructor(event)
 
         stereo_params = event.dl2.stereo.geometry["HillasReconstructor"]
 
-        if stereo_params.az < 0:
-            stereo_params.az += u.Quantity(360, u.deg)
+        if not stereo_params.is_valid:
+            logger.warning(
+                f"--> event {i_evt} (event ID {event_id}) failed to get valid stereo "
+                "parameters, possibly due to images of width = 0. Skipping..."
+            )
+            continue
+
+        stereo_params.az.wrap_at(360 * u.deg, inplace=True)  # Wrap at 0 <= az < 360 deg
 
         for tel_id in tel_ids:
 
-            # Calculate the impact parameter:
+            # Calculate the impact parameter
             impact = calculate_impact(
+                shower_alt=stereo_params.alt,
+                shower_az=stereo_params.az,
                 core_x=stereo_params.core_x,
                 core_y=stereo_params.core_y,
-                az=stereo_params.az,
-                alt=stereo_params.alt,
                 tel_pos_x=tel_positions[tel_id][0],
                 tel_pos_y=tel_positions[tel_id][1],
                 tel_pos_z=tel_positions[tel_id][2],
             )
 
-            # Set the stereo parameters to the data frame:
+            # Set the stereo parameters to the data frame
             params = {
-                "alt": stereo_params.alt.to(u.deg).value,
-                "alt_uncert": stereo_params.alt_uncert.to(u.deg).value,
-                "az": stereo_params.az.to(u.deg).value,
-                "az_uncert": stereo_params.az_uncert.to(u.deg).value,
-                "core_x": stereo_params.core_x.to(u.m).value,
-                "core_y": stereo_params.core_y.to(u.m).value,
-                "core_uncert": stereo_params.core_uncert.to(u.m).value,
```

Comment:
was core uncertainty dropped on purpose?

Reply:
Yes, it is because even if the stereo reconstruction is succeeded, the core uncertainty is not assigned to the container. Please see the definition of the [HillasReconstructor](https://github.com/cta-observatory/ctapipe/blob/22655d189d6fe1330e01891ff12296b621313bdf/ctapipe/reco/hillas_reconstructor.py#L295) in ctapipe (v0.12.0).

Reply:
And also the `h_max_uncert` is also not assigned, thus I dropped the parameter.

Reply:
thx for the explanation, I was precisely wondering that it was the only one dropped.I checked also the current version of ctapipe, and while it added some new fields (like the tilted frame core and mean intensity) those uncertainties does not seem to be there  still, so best to skip them as you did

Reply:
OK so just in case let me add a comment why we skip those parameters.